packages/boost-demo/src/App.tsx

```diff
@@ -82,49 +85,70 @@ function App() {
     setPayloadId(e.target.value);
   };
 
-  const makePayment = () => {
-    setErrorText("");
-    if (nitroClient && selectedChannel) {
-      nitroClient.Pay(selectedChannel, 100);
-      // TODO: Slightly hacky but we wait a beat before querying so we see the updated balance
-      setTimeout(() => {
-        updateChannelInfo(selectedChannel);
-      }, 50);
-    }
+  const updatePaymentAmount = (e: ChangeEvent<HTMLInputElement>) => {
+    setPaymentAmount(parseInt(e.target.value));
   };
 
-  const fetchFile = () => {
+  const handleUseBadSigChanged = (
+    _: ChangeEvent<HTMLInputElement>,
+    checked: boolean
+  ) => {
+    setUseBadSig(checked);
+  };
+  const fetchFile = async () => {
     setErrorText("");
 
-    axios
-      .get(
-        `http://localhost:7777/ipfs/${payloadId}?channelId=${selectedChannel}`,
+    if (!nitroClient) {
+      setErrorText("Nitro client not initialized");
+      return;
+    }
+    if (!selectedChannel) {
+      setErrorText("Please select a channel");
+      return;
+    }
+
+    const voucher = await nitroClient.CreateVoucher(
+      selectedChannel,
+      paymentAmount
+    );
+    // TODO: Slightly hacky but we wait a beat before querying so we see the updated balance
+    setTimeout(() => {
+      updateChannelInfo(selectedChannel);
+    }, 50);
+
+    const signatureToUse = useBadSig
+      ? generateRandomSignature()
+      : voucher.Signature;
+    try {
+      const result = await axios.get(
+        `http://localhost:7777/ipfs/${payloadId}?channelId=${voucher.ChannelId}&amount=${voucher.Amount}&signature=${signatureToUse}`,
         {
           responseType: "blob", // This lets us download the file
           headers: {
             Accept: "*/*", // TODO: Do we need to specify this?
           },
         }
-      )
-
-      .then((result) => {
-        // This will prompt the browser to download the file
-        const blob = result.data;
-        const blobUrl = URL.createObjectURL(blob);
-        const a = document.createElement("a");
-        a.href = blobUrl;
-        a.download = "fetched-file-from-ipfs";
-        a.click();
-        a.remove();
-        window.URL.revokeObjectURL(blobUrl);
-      })
-      .catch((e) => {
-        if (isAxiosError(e)) {
-          setErrorText(`${e.message}: ${e.response?.statusText}`);
-        } else {
-          setErrorText(JSON.stringify(e));
-        }
-      });
+      );
+
+      // This will prompt the browser to download the file
+      const blob = result.data;
+      const blobUrl = URL.createObjectURL(blob);
+      const a = document.createElement("a");
+      a.href = blobUrl;
+      a.download = "fetched-file-from-ipfs";
+      a.click();
+      a.remove();
+      window.URL.revokeObjectURL(blobUrl);
```

Comment:
Could possibly be factored into a `triggerFileDownload` helper.

Reply:
https://github.com/statechannels/nitro-gui/pull/88/commits/68879f70ac9ec8957a411877ac1494e54ba443c1